Core/Types/CkanModule.cs

```diff
@@ -366,6 +376,33 @@ private void DeSerialisationFixes(StreamingContext like_i_could_care)
             name = name ?? string.Empty;
         }
 
+        static CkanModule()
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            using (var stream = assembly.GetManifestResourceStream(SchemaPath))
+            using (var reader = new StreamReader(stream))
+            {
+                var result = reader.ReadToEnd();
+                MetadataSchema = JsonSchema4.FromSampleJson(result);
+            }
+        }
+
+        private static readonly string SchemaPath = "CKAN.CKAN.schema";
+        private static readonly JsonSchema4 MetadataSchema;
+
+        private static async Task<IList<string>> ValidateAgainstSchema(string json)
+        {
+            JsonSchema4 schema = await JsonSchema4.FromJsonAsync(json);
+
+            IList<string> errorList = new List<string>();
+            var errors = schema.Validate(MetadataSchema.ToJson());
```

Comment:
Actually, are you sure this call is correct? The `schema` object appears to be the module, and the parameter with ToJson is the schema. This looks like we're trying to use the module to validate the schema rather than the other way around. The [documentation](https://github.com/RSuter/NJsonSchema/wiki/JsonSchema4) passes the data to be validated to the schema's Validate function.